src/Development/Shake/FilePattern.hs

```diff
@@ -184,7 +184,7 @@ compatible (x:xs) = all ((==) (specials x) . specials) xs
 
 -- | Extract the items that match the wildcards. The pair must match with '?=='.
 extract :: FilePattern -> FilePath -> [String]
-extract p@(parse -> pat) = \x -> case match pat (split isPathSeparator x) of
+extract p@(parse -> pat) x = case match pat (split isPathSeparator x) of
```

Comment:
If that's the case I really need to revise my beliefs on how haskell works under the hood. I thought `f x y =` is just syntactic sugar for `f x = \y ->`